config-provisioning/src/main/java/com/yahoo/config/provision/NodeType.java

```diff
@@ -71,4 +71,11 @@ public NodeType childNodeType() {
         return childNodeTypes;
     }
 
+    /**
+     * @param type Child {@link NodeType}
+     * @return true if the {@link NodeType} can run on this host, false otherwise.
+     */
+    public boolean supportsChild(NodeType type) {
+        return isDockerHost() && childNodeTypes.contains(type);
```

Comment:
Can be simplified to just `childNodeTypes.contains(type)` as `isDockerHost` only checks the emptiness of `childNodeTypes`.